testing/upgradetest/upgrader.go

```diff
@@ -332,11 +332,15 @@ func PerformUpgrade(
 
 	upgradeOutput, err := startFixture.Exec(ctx, upgradeCmdArgs)
 	if err != nil {
+		logger.Logf("Upgrade err: %v, output: %s", err, upgradeOutput)
 		// Sometimes the gRPC server shuts down before replying to the command which is expected
 		// we can determine this state by the EOF error coming from the server.
 		// If the server is just unavailable/not running, we should not succeed.
 		// Starting with version 8.13.2, this is handled by the upgrade command itself.
 		outputString := string(upgradeOutput)
+		if strings.Contains(outputString, "upgrade did not occur because it is the same version") {
```

Comment:
Why is the check and the specific error needed here ?

Reply:
If we don't check the output here then our integration tests will wait for the watcher to start (which fails)